Makefile

```diff
@@ -1,5 +1,13 @@
 #! This is the root-level Makefile, which includes all the others
 
+# Custom variable to detect when Makefile is called through `emmake make`
+ifdef EMSCRIPTEN
+__EMSCRIPTEN__ = 1
+$(info 'EMSCRIPTEN' is defined, building for emscripten platform)
+else ifeq ($(OSMODE),emscripten)
+$(error You need to call 'emmake make' instead of simply 'make' if you want to build with emscripten. \
+	see https://emscripten.org/docs/compiling/Building-Projects.html#integrating-with-a-build-system)
+endif
```

Comment:
This is the very first lines of the Makefile because we need this define for the OBJDIR variable, which is itself above any include.
I don't like that I can't use our Makefile's print function, but I don't see cleaner solution either

Reply:
You can indeed use the Makefile's print_error function, but you need to suppres the output, so that make doesnt think that the command output is makefile code. I usually do this by assigning the output the to a useless variable, named underscore: `_`

Here is an example of this trick (this example comes from `rules/packages.mk`):
```mk

# if file doesn't exist, create it
ifeq ($(shell test -f $(PACKAGESFILE) ; echo $$?),1)
_:=$(shell $(call print_warning,"packages list file '$(PACKAGESFILE)' doesn't exist - creating now..."))
_:=$(shell mkdir -p `dirname $(PACKAGESFILE)`)
_:=$(shell $(call make_PACKAGESFILE))
_:=$(shell mkdir -p $(PACKAGESDIR))
endif
```

Reply:
I don't think I can, given that the print_* functions are included after this definition